src/Microsoft.ML.Data/DataView/Transposer.cs

```diff
@@ -1536,113 +1523,50 @@ public SlotDataView(IHostEnvironment env, ITransposeDataView data, int col)
-
-                public SchemaImpl(SlotDataView parent)
-                {
-                    Contracts.AssertValue(parent);
-                    _parent = parent;
-                    AsSchema = Schema.Create(this);
-                }
-
-                public ColumnType GetColumnType(int col)
-                {
-                    Host.CheckParam(col == 0, nameof(col));
-                    return _parent._type;
-                }
-
-                public string GetColumnName(int col)
-                {
-                    Host.CheckParam(col == 0, nameof(col));
-                    // There is no real need for this to have the real name as the internal IDV
-                    // substream does not have its name accessed, but we'll save it just the same.
-                    // I am tempted though to just have this thing always claim its name is 'Pancakes'.
-                    return _parent._data.Schema.GetColumnName(_parent._col);
-                }
-
-                public bool TryGetColumnIndex(string name, out int col)
-                {
-                    if (name == GetColumnName(0))
-                    {
-                        col = 0;
-                        return true;
-                    }
-                    col = -1;
-                    return false;
-                }
-
-                // No metadata. The top level IDV will hold the schema information, including metadata.
-                // This per-column dataview schema information is just minimally functional.
-
-                public IEnumerable<KeyValuePair<string, ColumnType>> GetMetadataTypes(int col)
-                {
-                    Host.CheckParam(col == 0, nameof(col));
-                    return Enumerable.Empty<KeyValuePair<string, ColumnType>>();
-                }
-
-                public ColumnType GetMetadataTypeOrNull(string kind, int col)
-                {
-                    Host.CheckNonEmpty(kind, nameof(kind));
-                    Host.CheckParam(col == 0, nameof(col));
-                    return null;
-                }
-
-                public void GetMetadata<TValue>(string kind, int col, ref TValue value)
-                {
-                    Host.CheckNonEmpty(kind, nameof(kind));
-                    Host.CheckParam(col == 0, nameof(col));
-                    throw MetadataUtils.ExceptGetMetadata();
-                }
-            }
-
-            private sealed class Cursor<T> : SynchronizedCursorBase<ISlotCursor>, IRowCursor
+            private sealed class Cursor<T> : RootCursorBase
             {
                 private readonly SlotDataView _parent;
+                private readonly SlotCursor _slotCursor;
                 private readonly Delegate _getter;
 
-                public Schema Schema => _parent.Schema;
+                public override Schema Schema => _parent.Schema;
+
+                public override long Batch => throw new NotImplementedException();
 
```

Comment:
double-checking that this is intended.

Reply:
Maybe comment if so. 

---
In reply to: [238816862](https://github.com/dotnet/machinelearning/pull/1814#discussion_r238816862) [](ancestors = 238816862)

Reply:
Um no it was not intended, thank you!